docs/data/date-pickers/custom-components/custom-components.md

```diff
@@ -35,6 +35,11 @@ As an example, you could override the `ActionBar` and pass additional props to t
 />
 ```
 
+:::success
+Remember to pass a reference to the component instead of an inline render function and define it outside of the main component.
+This ensures that the component is not re-rendered on every update.
+:::
```

Comment:
This code is buggy because it will be a new input after each keystroke, leading to the loss of focus.
```jsx
function MyApp (){
  const [name, setName] = React.useState("");

  // This component get redefined each time `name` is updated ❌
  const CustomActionBar = () => <input 
    value={name} 
    onChange={event => setName(event.target.value)} 
  />;

  <DatePicker
    slots={{
      actionBar: CustomActionBar,
    }}
}
```

This one is correct since it's always the same input with different props.
```jsx
// This component is defined one time only ✅
const CustomActionBar = ({name, setName}) => <input 
  value={name} 
  onChange={event => setName(event.target.value)} 
/>;

function MyApp (){
  const [name, setName] = React.useState("");

  <DatePicker
    slots={{
      actionBar: CustomActionBar,
    }}
    slotProps={{
      actionBar: { name, setName},
    }}
}
```

Reply:
I was thinking a minimal example could help but maybe it's too long

Reply:
Makes sense, examples usually speak a thousand words.
Maybe a separate section would work here... 🤔

Reply:
I've created a separate section and added a bit more elaborate examples.
FYI, the examples are missing some empty lines in order to avoid the introduction of a scrollbar.
What do you think?

Reply:
The spacing did not surprised me 👍